packages/shared/src/lib/auxiliary/wallet-connect/constants/methods-for-permissions.constant.ts

```diff
@@ -1,7 +1,7 @@
 import { DappPermission } from '../enums/dapp-permission.enum'
 
 export const METHODS_FOR_PERMISSION = {
-    [DappPermission.SignData]: ['eth_sign', 'personal_sign', 'eth_signTypedData'],
+    [DappPermission.SignData]: ['eth_sign', 'personal_sign', 'eth_signTypedData', 'eth_signTypedData_v4'],
```

Comment:
this implementation also supports `v3`, so we can add it here.
```suggestion
    [DappPermission.SignData]: ['eth_sign', 'personal_sign', 'eth_signTypedData_v3', 'eth_signTypedData_v4'],
```